src/Aquifer.Common/Utilities/DictionaryExtensions.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Aquifer.Common.Utilities;
+
+public static class DictionaryExtensions
+{
+    public static TValue? GetValueOrNull<TKey, TValue>(this IDictionary<TKey, TValue> source, TKey key)
```

Comment:
`GetValueOrDefault()` doesn't support returning `null` for a `Dictionary<int, int>` so I wrote this extension method to allow null chaining.